### PR TITLE
Bandchain.js: Add getReferenceData

### PR DESCRIPTION
**Pull request by Pitchanai**
## Implementation details
- Add getReferenceData

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [x] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [x] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

**Review by pzshine — APPROVED** [review]

File: helpers/bandchain2.js/src/client.ts [hunk 1]
```diff
@@ -156,6 +161,52 @@ export default class Client {
     }
   }
 
+
+  async getReferenceData(pairs: string[]): Promise<any[]> {
+    let symbolSet: Set<string> = new Set()
+    pairs.forEach((pair) => {
```
Comment (pzshine) [inline helpers/bandchain2.js/src/client.ts]: Can you create the interface for this function?

**Review by pzshine — CHANGES_REQUESTED** [review]

**Review by pzshine — APPROVED** [review]

File: helpers/bandchain2.js/__tests__/client/client.test.ts [hunk 2]
```diff
@@ -50,6 +50,55 @@ describe('Client get data', () => {
...
+      },
+    }
+    mockedAxios.post.mockResolvedValue(resp)
+
+    const response = client.getReferenceData(['BTC/USD', 'TRX/ETH'])
+    const expected = [
+      {
+        pair: 'BTC/USD',
+        rate: 16242.693800000001,
+        updated: { base: 1605512243, quote: Math.round(Date.now() / 1000) },
```
Comment (evilpeach) [inline helpers/bandchain2.js/__tests__/client/client.test.ts]: Math.floor?

File: helpers/bandchain2.js/src/client.ts [hunk 3]
```diff
@@ -156,6 +162,51 @@ export default class Client {
     }
   }
 
+  async getReferenceData(pairs: string[]): Promise<ReferenceData[]> {
+    let symbolSet: Set<string> = new Set()
+    pairs.forEach((pair: string) => {
+      let symbols = pair.split('/')
+      symbols.forEach((symbol: string) => {
+        if (symbol == 'USD') return
```
Comment (evilpeach) [inline helpers/bandchain2.js/src/client.ts]: Please use the strict equality operator `===`

File: helpers/bandchain2.js/src/client.ts [hunk 4]
```diff
@@ -156,6 +162,51 @@ export default class Client {
...
+      ask_count: 4,
+    }
+    let priceData = await this.postResult('/oracle/request/prices', pricerBody)
+
+    let symbolMap: any = {}
+    symbolMap['USD'] = {
+      symbol: 'USD',
+      multiplier: '1000000000',
+      px: '1000000000',
+      resolve_time: Math.round(Date.now() / 1000).toString(),
```
Comment (evilpeach) [inline helpers/bandchain2.js/src/client.ts]: Math.floor?

File: helpers/bandchain2.js/src/client.ts [hunk 5]
```diff
@@ -156,6 +162,51 @@ export default class Client {
...
+    })
+    let data: ReferenceData[] = []
+    pairs.forEach((pair) => {
+      let [baseSymbol, quoteSymbol] = pair.split('/')
+
+      data.push({
+        pair,
+        rate:
+          (symbolMap[baseSymbol].px * symbolMap[quoteSymbol].multiplier) /
+          (symbolMap[quoteSymbol].px * symbolMap[baseSymbol].multiplier),
```
Comment (evilpeach) [inline helpers/bandchain2.js/src/client.ts]: Let's parse to integer before dividing

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**